Orange/widgets/utils/domaineditor.py

```diff
@@ -80,7 +80,8 @@ def setData(self, index, value, role):
         row_data = self.variables[row]
         if role == Qt.EditRole:
             if col == Column.name:
-                row_data[col] = value
+                if not (value.isspace() or value == ""):
```

Comment:
This results in the above `if col == Column.name` succeeding and then the inside `if` doing nothing for empty values. But the function still returns True in the end.
Instead, using `if col == Column.name and not (value.isspace() or value == "")` would result either in the value being updated or nothing happens and function returns False. Which means that after an "incorrect" edit the Apply button does not become enabled (same as cancelling the edit with Esc).